File: 821_Shortest_Distance_to_a_Character/sol.py
class Solution:
    def shortestToChar(self, S: str, C: str) -> List[int]:
        prev = int(-10001)
        res = []
        for i, c in enumerate(S):
            if c == C:
                prev = i
            res.append(i-prev)

        prev = int(10001)
        for i in range(len(S) -1, -1, -1):
            if S[i]==C:
                prev = i
            res[i] = min(res[i], prev-i)
            
            
        return res

File: 821_Shortest_Distance_to_a_Character/sol.cpp
class Solution {
public:
    vector<int> shortestToChar(string S, char C) {
        auto prev = -10001;
        vector<int> res;
        for(int i=0; i < S.size(); i++){
            if (S[i]==C)
                prev = i;
            res.push_back(i - prev);
        }
        
        prev = 10001;
        for(int i=S.size()-1; i>=0; i--){
            if (S[i]==C)
                prev = i;
            res[i] = std::min(res[i], prev - i);
        }
        
        return res;
    }
};
